autoscheduler/scraper/tests/api_tests.py

```diff
@@ -318,3 +326,210 @@ def test_api_course_search_gives_correct_results_c(self):
+
+        # Act
+        response = self.client.get('/api/course/search', data=data)
+
+        # Assert
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.json(), expected)
+
+    def test_api_course_search_gives_correct_results_csce_lower(self):
+        """ Tests that /api/course/search gives the correct response for a search
+            containing only lowercase letters
+        """
+        # Arrange
+        expected = {'results': ['CSCE 181', 'CSCE 310', 'CSCE 315']}
+        data = {'search': 'csce', 'term': '201731'}
+
+        # Act
+        response = self.client.get('/api/course/search', data=data)
+
+        # Assert
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.json(), expected)
+
+    def test_api_course_search_gives_correct_results_csce_3_lower(self):
+        """ Tests that /api/course/search gives the correct response for a search
+            containing lowercase letters and a number
+        """
+        # Arrange
+        expected = {'results': ['CSCE 310', 'CSCE 315']}
+        data = {'search': 'csce%203', 'term': '201731'}
+
+        # Act
+        response = self.client.get('/api/course/search', data=data)
+
+        # Assert
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.json(), expected)
+
+    def test_api_course_search_gives_correct_results_csce_space_310_lower(self):
+        """ Tests that /api/course/search gives the correct response for a search
+            containing a space not already in %20
+        """
+        # Arrange
+        expected = {'results': ['CSCE 310']}
+        data = {'search': 'csce 310', 'term': '201731'}
+
+        # Act
+        response = self.client.get('/api/course/search', data=data)
+
+        # Assert
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.json(), expected)
+
+    def test_api_term_serializer_gives_expected_output_non_professional(self):
+        """ Tests that the term serializer yields the correct data for
+            a non professional term
+        """
+        # Arrange
+        expected = {
+            'term' : '201831',
+            'desc': 'Fall 2018 - College Station'
+        }
+        # Save department for testing
+        dept = Department(id='CSCE201831', code='CSCE', term='201831')
+        dept.save()
+
+        # Act
+        serializer = TermSerializer(dept)
+
+        # Assert
+        self.assertEqual(expected, serializer.data)
+
+    def test_api_term_serializer_gives_expected_output_professional(self):
+        """ Tests that the term serializer yields the correct data for
+            a professional term
+        """
+        # Arrange
+        expected = {
+            'term' : '201941',
+            'desc': 'Full Yr Professional 2019 - 2020'
+        }
+        # Save department for testing
+        dept = Department(id='DDDS201941', code='DDDS', term='201941')
+        dept.save()
```

Comment:
You don't actually have to save models to serialize them (also remove the "save department" comment)
```suggestion
```